src/main/java/view25d/view/TurtleView.java

```diff
@@ -100,20 +108,86 @@ public void manuallyRefreshReporterView(Context context)  throws LogoException {
 
     private void updateArrayList(Context context) throws LogoException {
         //turtleReporterValues.clear();
-        turtleReporterValues = new ArrayList<TurtleValue>();
+
+        ArrayList<TurtleValue> tempTurtles = new ArrayList<TurtleValue>();
+        ArrayList<LinkValue> tempLinks = new ArrayList<LinkValue>();
+
         for (Agent a : myAgents.agents()) {
             Turtle turtle = (Turtle)a;
+            if (turtle.hidden()) {
+                continue;
+            }
             Color c = org.nlogo.api.Color.getColor(turtle.color());
             double val = (Double)reporter.report(context, new Object[]{turtle});
             double stemColor = getStemColor(context, turtle);
             TurtleValue tv = new TurtleValue( turtle.shape(), c, turtle.size(), turtle.xcor(), turtle.ycor(), val, stemColor);
-            turtleReporterValues.add(tv);
+            tempTurtles.add(tv);
         }
+
+        // Get Set of Links associated with the Turtles
+        Set<Link> linkSet = getLinkSetFromTurtleSet(myAgents);
+
+        // Store the data needed for link visualization
+        for (Link link : linkSet) {
+
+            Turtle end1 = (Turtle)link.end1();
+            Turtle end2 = (Turtle)link.end2();
+
+            Color c = org.nlogo.api.Color.getColor(link.color());
+            double zcor1 = (Double)reporter.report(context, new Object[]{end1});
+            double zcor2 = (Double)reporter.report(context, new Object[]{end2});
+            
+            LinkValue lv = new LinkValue(link.shape(), c, link.lineThickness(),
+                                         end1.xcor(), end1.ycor(), zcor1,
+                                         end2.xcor(), end2.ycor(), zcor2);
+            tempLinks.add(lv);
+        }
+
+        turtleReporterValues = tempTurtles;
+        linkValues = tempLinks;
+            
         if (viewOptions.usePColor()) {
             updatePColors();
+        }  
+    }
+
+
+    // Given an AgentSet of turtles, produce a Set of associated Links.
+    // Assumes check that AgentSet contains Turtles has already been made.
+    // Currently includes links for which only one end is in the AgentSet
+    public static  Set<Link>  getLinkSetFromTurtleSet(AgentSet turtleSet) {
+
+        Set<Link> linkSet = new HashSet<>();
+        
+        // no work to do if there are no links or no turtles
+
+        World world = App.app().workspace().world();
+        
+        if (world.links().count() == 0 || turtleSet.count() == 0) {
+            return linkSet];
```

Comment:
I think a `]` slipped in here that wasn't intended.  I removed it and then was able to package and run the extension with no problems.